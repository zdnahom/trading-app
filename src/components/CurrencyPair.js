import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight, faClock } from '@fortawesome/free-solid-svg-icons';
import '../styles/CurrencyPair.css';

const CurrencyPair = ({ exchangeItem }) => (
  <li className="currency-card">
    <FontAwesomeIcon icon={faCircleArrowRight} className="forward-icon" />
    <div className="currency-card-content">
      <div className="currency-rate">
        Bid
        <span>{exchangeItem.bid}</span>
        Ask
        <span>{exchangeItem.ask}</span>
      </div>
      <div className="currency-pair-container">
        <h3 className="currency-pair-text">{exchangeItem.ticker}</h3>
        <span className="currency-pair-time-container">
          <FontAwesomeIcon icon={faClock} />
          04:54:09
        </span>
        <span>Currency</span>
      </div>
    </div>
  </li>
);

export default CurrencyPair;
CurrencyPair.propTypes = {
  exchangeItem: PropTypes.shape({
    ticker: PropTypes.string.isRequired,
    bid: PropTypes.string.isRequired,
    ask: PropTypes.string.isRequired,
    open: PropTypes.string.isRequired,
    low: PropTypes.string.isRequired,
    high: PropTypes.string.isRequired,
    changes: PropTypes.number.isRequired,
    date: PropTypes.string.isRequired,
  }).isRequired,
};
