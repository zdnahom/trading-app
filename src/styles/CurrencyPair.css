.currency-card {
  list-style: none;
}

.currency-card button {
  background-color: #4369b2;
  border: none;
  height: 150px;
  width: 100%;
  display: flex;
  flex-direction: column;
  color: white;
  padding: 2% 3%;
  cursor: pointer;
}

.forward-icon {
  align-self: end;
}

.currency-card-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 100%;
}

.currency-rate {
  display: flex;
  flex-direction: column;
  color: #d2d2d2;
  align-items: center;
}

.currency-rate span {
  font-size: 1.1rem;
  font-weight: 700;
  color: white;
  margin-bottom: 0.4rem;
}

.currency-pair-container {
  display: flex;
  flex-direction: column;
  gap: 0.4rem;
  align-items: center;
}

.currency-pair-text {
  font-size: 1.4rem;
  font-weight: 700;
}

.currency-pair-time-container {
  display: flex;
  gap: 0.4rem;
  align-items: center;
  color: rgb(169, 238, 169);
}
