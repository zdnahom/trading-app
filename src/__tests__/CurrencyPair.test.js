import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import CurrencyPair from '../components/CurrencyPair';

const exchangeData = {
  ticker: 'EUR/USD',
      bid: '1.09734',
      ask: '1.09734',
      open: '1.09720',
      low: '1.09680',
      high: '1.09755',
      changes: 0.000140000000000029,
      date: '2023-04-18 18:48:20',
};

describe('CurrencyPair component test', () => {
  const handleFuncion = () => {
    return "Got to detail page"
  }
  it('Currency pair component matches the snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <CurrencyPair exchangeItem={exchangeData} handleSelect={handleFuncion}/>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('It renders correctly', () => {
    render(
      <BrowserRouter>
          <CurrencyPair exchangeItem={exchangeData} handleSelect={handleFuncion}/>
      </BrowserRouter>,
    );
    const pair = screen.getByText('EUR/USD');
    expect(pair).toBeInTheDocument();
  });
});
