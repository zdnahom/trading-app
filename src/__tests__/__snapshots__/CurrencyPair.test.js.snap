// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyPair component test Currency pair component matches the snapshot 1`] = `
<div>
  <a
    href="/EUR_USD"
    style="text-decoration: none;"
  >
    <li
      class="currency-card"
    >
      <button
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-arrow-right forward-icon"
          data-icon="circle-arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
            fill="currentColor"
          />
        </svg>
        <div
          class="currency-card-content"
        >
          <div
            class="currency-rate"
          >
            Bid
            <span>
              1.09734
            </span>
            Ask
            <span>
              1.09734
            </span>
          </div>
          <div
            class="currency-pair-container"
          >
            <h3
              class="currency-pair-text"
            >
              EUR/USD
            </h3>
            <span
              class="currency-pair-time-container"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-clock "
                data-icon="clock"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                  fill="currentColor"
                />
              </svg>
              04:54:09
            </span>
            <span>
              Currency
            </span>
          </div>
        </div>
      </button>
    </li>
  </a>
</div>
`;
